gulpTasks/definition.js

```diff
@@ -88,4 +89,13 @@ gulp.task('internalDefs', function() {
   });
 });
 
-gulp.task('validateDefs', shell.task(['node node_modules/typescript/bin/tsc --noEmit ./bin/ts/CoveoJsSearch.d.ts']));
+gulp.task('validateDefs', ['validateTSV1', 'validateTSV2']);
+
+gulp.task('installTSV1', done => {
```

Comment:
Once that task has been executed, I'm guessing that any task that we execute after this one will now be using version 1.8 of typescript, meaning that we need to ensure it always get executed last in the pipeline.

Any way the task could clean itself up after it's done doing it's job ? Like re-setting typescript to whatever version it was initially.

Reply:
It installs the specified typescript version in a directory inside the `tvm` node_modules. Running `gulp` after using this task shows that the `2.7.1` typescript version so we should be fine.